mono/metadata/assembly.c

```diff
@@ -2838,7 +2836,7 @@ mono_assembly_binding_applies_to_image (MonoAssemblyLoadContext *alc, MonoImage*
 	MonoAssembly *result_ass = NULL;
 	MonoAssemblyName *result_name = &probed_aname;
 	result_name = mono_assembly_apply_binding (result_name, &dest_name);
-	if (result_name != &probed_aname && !mono_assembly_names_equal (result_name, &probed_aname)) {
+	if (result_name != &probed_aname && !mono_assembly_check_name_match (result_name, &probed_aname)) {
```

Comment:
Not sure this change is needed.  netcore shouldn't get here (no binding redirections).  I think it should stay an eq check.